### ISSUE TITLE
communicability_betweenness_centrality error (probably matrix/vector dimension error)

### ISSUE THREAD

**Issue by zericardo** [networkx/networkx#2364]
Hello everyone, I'm trying to compute the communicability_betweenness_centrality for a Barabási-Albert graph, with the following code (networkx version 1.11, python3):

```python
import networkx as nx

G = nx.barabasi_albert_graph(1000, 3)
c_bet = nx.communicability_betweenness_centrality(G)
```

However, it raises the following error:

```python
/home/zr/anaconda3/lib/python3.5/site-packages/scipy/sparse/linalg/matfuncs.py in _rmatvec(self, x)
    201         x = x.ravel()
    202         for i in range(self._p):
--> 203             x = A_T.dot(x)
    204         return x
    205 

ValueError: shapes (1000,1000) and (1,1000) not aligned: 1000 (dim 1) != 1 (dim 0)
```
I did the same thing with the karate club network and it worked normally. 

Thank you.

**Comment by peterewills** [comment]
The issue is somewhere in `scipy.linalg.expm`. This function breaks when applied to **adjacency matrices of size greater than or equal to 200**. For some reason this only happens with the input to `expm` is a numpy matrix. It's not a problem for numpy arrays. For example,

    >>> import numpy as np
    >>> from scipy.linalg import expm
    >>> A = np.diag(np.ones(199,1)) + np.diag(np.ones(199,-1)) # a path graph
    >>> expm(A) # works fine
    >>> expm(np.matrix(A)) # breaks

This problem is not present for all matrices of size over 200. For example, `expm(np.matrix(np.eye(200)))` works fine.

A (somewhat cheesy) way to avoid the problem is to convert the adjacency matrix to an array before applying `expm`.

**Comment by peterewills** [comment]
It looks like `communicability_betweenness_centraility` isn't present in networkx version 2.0. Should I add it back in with the above changes? Or is there another reason it was removed?

Apologies if this is a stupid question, this is my first time contributing to an open-source project.

**Comment by hagberg** [comment]
It looks like https://github.com/networkx/networkx/blob/master/networkx/algorithms/centrality/subgraph_alg.py#L242
need to be changed to say `expm(A.A)`

**Comment by hagberg** [comment]
Fixed in #2372
